src/main/java/datawave/query/util/AllFieldMetadataHelper.java

```diff
@@ -1061,350 +1059,309 @@ public Set<String> loadDatatypes() throws TableNotFoundException {
     /**
      * Fetches results from {@link #metadataTableName} and calculates the set of field index holes that exists for all indexed entries. The map consists of
      * field names to datatypes to field index holes.
-     * 
+     *
+     * @param minThreshold
+     *            the minimum percentage threshold required for an index row to be considered NOT a hole on a particular date
      * @return a map of field names and datatype pairs to field index holes
      */
     @Cacheable(value = "getFieldIndexHoles", key = "{#root.target.auths,#root.target.metadataTableName}", cacheManager = "metadataHelperCacheManager")
-    public Map<String,Map<String,FieldIndexHole>> getFieldIndexHoles() throws TableNotFoundException, CharacterCodingException {
-        return getFieldIndexHoles(ColumnFamilyConstants.COLF_I);
+    public Map<String,Map<String,FieldIndexHole>> getFieldIndexHoles(int minThreshold) throws TableNotFoundException, IOException {
```

Comment:
I am not sure that an integer percentage is enough granularity.  Can we make this a float/double beween 0.0 and 1.0 ?

Reply:
Done, it's now a double that must be between 0.0-1.0.